### PR TITLE
Feign client

### PR DESCRIPTION
**Pull request by johnsonr**
The Feign work is unfinished, but it's important to get it in so as to avoid merge conflicts.

This PR

- adds incomplete Feign functionality
- improves some core Java functionality and Spring functionality
- improves the fake `Project` support
- changes the intent from `generate` to `create project`: Discussed that with @WillGardella yesterday and thought I'd pushed it but I hadn't

### CODE REVIEW

File: .atomist/common/java/JavaGrammars.ts [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+ */
 export const DISCARDED_ANNOTATION = Microgrammar.fromDefinitions("discardAnnotations", {
     _at: "@",
     _annotationName: CLASS_NAME,
     _content: new Opt(JavaParenthesizedExpression),
-    ...AnonymousDefinition,
+    ...AnonymousDefinition
+});
+
+export const SPRING_BOOT_APP = Microgrammar.fromDefinitions<{name: string}>("pkg", {
```
Comment (ddgenome) [inline .atomist/common/java/JavaGrammars.ts]: This is here: https://github.com/atomisthq/spring-team-handlers/blob/master/.atomist/common/java/spring/SpringGrammars.ts#L8

File: .atomist/common/java/spring/SpringBootProjectStructure.ts [hunk 2]
```diff
@@ -13,37 +11,49 @@ export class SpringBootProjectStructure {
         const appFiles = p.files
             .filter(f => /\.java$/.test(f.filename))
             .filter(f => f.content.indexOf("@SpringBootApplication") > -1);
-        if (appFiles.length === 0) {
```
Comment (ddgenome) [inline .atomist/common/java/spring/SpringBootProjectStructure.ts]: Would be good to retain these changes.

File: .atomist/common/java/spring/SpringBootProjectStructure.ts [hunk 3]
```diff
@@ -13,37 +11,49 @@ export class SpringBootProjectStructure {
...
+        if (appFiles.length == 0)
             return null;
-        }
-        if (appFiles.length > 1) {
+        if (appFiles.length > 1)
```
Comment (ddgenome) [inline .atomist/common/java/spring/SpringBootProjectStructure.ts]: Retain

File: .atomist/common/java/JavaGrammars.ts (hunk 1 above)
Comment (johnsonr) [inline .atomist/common/java/JavaGrammars.ts]: Thanks. Fixed

File: .atomist/common/java/spring/SpringBootProjectStructure.ts (hunk 3 above)
Comment (johnsonr) [inline .atomist/common/java/spring/SpringBootProjectStructure.ts]: Done

**Pull request merged**